src/Integration.Vsix/SonarAnalyzerManager.cs

```diff
@@ -37,52 +44,210 @@ public sealed class SonarAnalyzerManager
             DifferentVersion
         }
 
+        private readonly IServiceProvider serviceProvider;
         private readonly Workspace workspace;
         private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+        private readonly ILogger logger;
+        private readonly ISonarQubeService sonarQubeService;
 
         private static readonly AssemblyName AnalyzerAssemblyName =
             new AssemblyName(typeof(SonarAnalysisContext).Assembly.FullName);
         internal /*for testing purposes*/ static readonly Version AnalyzerVersion = AnalyzerAssemblyName.Version;
         internal /*for testing purposes*/ static readonly string AnalyzerName = AnalyzerAssemblyName.Name;
 
-        internal /*for testing purposes*/ SonarAnalyzerManager(IActiveSolutionBoundTracker activeSolutionBoundTracker,
-            Workspace workspace)
+        private DelegateInjector delegateInjector;
+        private ISonarQubeIssuesProvider sonarqubeIssueProvider;
+        private SuppressionHandler suppressionHandler;
+
+        public SonarAnalyzerManager(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            this.serviceProvider = serviceProvider;
+
+            this.activeSolutionBoundTracker = serviceProvider.GetMefService<IActiveSolutionBoundTracker>();
+            Debug.Assert(this.activeSolutionBoundTracker != null, "Cannot retrieve an object of type " +
+                $"{nameof(IActiveSolutionBoundTracker)} from MEF catalog.");
+            this.activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
+
+            this.logger = serviceProvider.GetMefService<ILogger>();
+            Debug.Assert(this.logger != null, $"Cannot retrieve an object of type {nameof(ILogger)} from MEF catalog.");
+
+            var componentModel = serviceProvider.GetService<SComponentModel, IComponentModel>();
+            this.workspace = componentModel.GetService<VisualStudioWorkspace>();
```

Comment:
Just use _serviceProvider.GetMefService<VisualStudioWorkspace>_ (it's just a wrapper that fetches the componentModel for you).